File: barney.py
isSad = True

class SadBarney:
	def stop(self):
		print "I stop being sad"

def sad():
    global isSad
    if (isSad):
        isSad = False
        print "When I get sad,"
        return True
    else:
        return SadBarney()

def beAwesome():
    print "and be awesome instead"
    
#life motto
if (sad() == True): 
    sad().stop()
    beAwesome()

File: barney.php
<?php
$isSad = true;

class SadBarney {
    public function stop() {
        echo "I stop being sad,\n";
    }
}

function sad() {
    global $isSad;
    if ($isSad) {
        $isSad = false;
        echo "When I get sad\n";
        return true;
    } 
    else return new SadBarney();
}

function beAwesome() {
    echo "and be awesome instead\n";
}

//life moto
if (sad() === true) {
    sad()->stop();
    beAwesome();
}
